src/hooks/useHubspotForm.js

```diff
@@ -0,0 +1,103 @@
+/* @flow */
+
+import { useCallback, useState } from 'react';
+
+type FormConfig = {|
+  formGuid: string,
+  pageName: string,
+  pageUri: string,
+  portalId: string,
+|};
+
+type FormData = {
+  /*
+   * Hubspot's field name (same as the `input` `name` attribute on Hubspot Forms)
+   *
+   * e.g. { 'company_name': 'ABC', 'company_size': '6' }
+   */
+  [fieldName: string]: string,
+};
+
+type HubspotFieldsData = Array<{
+  name: string,
+  value: string,
+}>;
+
+type Status = {| error: string |} | {| response: Object |} | void;
+
+type OnError = () => void;
+type OnSuccess = () => void;
+
+type HookReturn = {
+  error: ?string,
+  response: ?Object,
+  submitForm: (
+    formData: FormData,
+    onSuccess?: OnSuccess,
+    onError?: OnError,
+  ) => void,
+};
+
+const useHubspotForm = ({
+  formGuid,
+  pageName,
+  pageUri,
+  portalId,
+}: FormConfig): HookReturn => {
+  // eslint-disable-next-line max-len
+  const endpoint = `https://api.hsforms.com/submissions/v3/integration/submit/${portalId}/${formGuid}`;
+
+  const [status, setStatus] = useState<Status>();
+
+  const prepareFieldsData = useCallback(
+    (formData: FormData): HubspotFieldsData =>
+      Object.keys(formData).map(fieldName => ({
+        name: fieldName,
+        value: formData[fieldName],
+      })),
+    [],
+  );
+
+  const submitForm = useCallback(
+    (formData: FormData, onSuccess?: OnSuccess, onError?: OnError) => {
+      const fetchOptions = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
```

Comment:
I'm not sure what Hubspot expects, but if we bothered to set the content type, shouldn't we also set the content encoding ? _(ie: `utf-8`)_